### ISSUE TITLE
ESPHome HiGrow plant sensor

### ISSUE THREAD

**Issue by BeardedTinker** [BeardedTinker/Home-Assistant_Config#74]
Playing with LilyGo Higrow board but this time using ESPHome as firmware for the board.

**Comment by Iananan** [comment]
do you have any insight on the BME280 version in ESPhome? using the config from here: https://esphome.io/components/sensor/bme280.html gives me 

```
[C][bme280.sensor:143]: BME280:
[C][bme280.sensor:144]:   Address: 0x77
[E][bme280.sensor:147]: Communication with BME280 failed!
```
in the logs.

it looks like the BME280 and BH1750 share the I2C_bus on 25 and 26 according to the pinouts for the BME280 version:
![image](https://user-images.githubusercontent.com/1507624/152641686-233b0a33-9f33-4a33-8b08-176459fda58e.png)

but adding i2c_id:  bus_a doesn't help, and attempting to give them each their own i2c_id causes both the BME280 and BH1750 to fail on boot. I've tried switching to BMP280 specs, but that isn't helping either

**Comment by BeardedTinker** [comment]
As I only have 1 board with BME280 and 6 with DHT, so far I've only been playing with DHT version.
But hopefully, I'll jump to that next. 
Did you try address 0x76 as alternative address?

**Comment by BeardedTinker** [comment]
> do you have any insight on the BME280 version in ESPhome? using the config from here: https://esphome.io/components/sensor/bme280.html gives me
> 
> ```
> [C][bme280.sensor:143]: BME280:
> [C][bme280.sensor:144]:   Address: 0x77
> [E][bme280.sensor:147]: Communication with BME280 failed!
> ```
> 
> in the logs.
> 
> it looks like the BME280 and BH1750 share the I2C_bus on 25 and 26 according to the pinouts for the BME280 version: ![image](https://user-images.githubusercontent.com/1507624/152641686-233b0a33-9f33-4a33-8b08-176459fda58e.png)
> 
> but adding i2c_id: bus_a doesn't help, and attempting to give them each their own i2c_id causes both the BME280 and BH1750 to fail on boot. I've tried switching to BMP280 specs, but that isn't helping either

As I only have 1 board with BME280 and 6 with DHT, so far I've only been playing with DHT version.
But hopefully, I'll jump to that next.
Did you try address 0x76 as alternative address?

**Comment by Iananan** [comment]
Ah, I'm new to these boards so got 3 BME boards to start with. I have tried 0x76 as an alternate, but after digging into it a bunch more, it seems like ESPhome really struggles with the BME280. Possibly down to its implementation of the wire.h and TwoWire Functions but I don't know enough C++ to really troubleshoot.

The chip itself is functioning fine as I've tested the Mqtt builds too and they output the temp/pressure/humidity perfectly, but I then struggled to get the auto discover working within containerised HA core. I'm also using ESPhome to bring in the miflora plant monitors data so ESPhome is just neatest all round. 

Tldr; definitely seems to be an issue with how ESPhome handles the bme280 afaict

**Comment by BeardedTinker** [comment]
> Ah, I'm new to these boards so got 3 BME boards to start with. I have tried 0x76 as an alternate, but after digging into it a bunch more, it seems like ESPhome really struggles with the BME280. Possibly down to its implementation of the wire.h and TwoWire Functions but I don't know enough C++ to really troubleshoot.
> 
> The chip itself is functioning fine as I've tested the Mqtt builds too and they output the temp/pressure/humidity perfectly, but I then struggled to get the auto discover working within containerised HA core. I'm also using ESPhome to bring in the miflora plant monitors data so ESPhome is just neatest all round.
> 
> Tldr; definitely seems to be an issue with how ESPhome handles the bme280 afaict

I've finished just now with new ESPHome for non BME version and will start to play with BME280 version in next few days.
There is issue with not specifying `setup_priority` on those boards. Not sure if you tried with it or not.

**Comment by BeardedTinker** [comment]
I've added support for MQTT too.
You can disable mqtt section if you don't need it

```
mqtt:
  broker: mqtt_ip
  username: mqtt_user
  password: mqtt_password
```

But if you do use it, in secrets file just add following 3 lines (or disable with comment # mqtt_user and mqtt_password if don't sue them):

```
mqtt_ip: 192.168.1.xxx
mqtt_user: xxxxxxxxxx
mqtt_password: xxxxxxxxxx
```
